### ISSUE TITLE
Windows 11 update: FAILED

### ISSUE THREAD

**Issue by GotAnAccount** [topgrade-rs/topgrade#482]
Looking in the code, you seem to be executing the update command twice. Is this normal?

## Erroneous Behavior
With a vanilla installation of topgrade through scoop, I keep getting a failed Windows update.

## Expected Behavior
I would expect that topgrade do start a "Windows Update" process.

## Steps to reproduce
open a command prompt and execute topgrade.

## Problem persists without calling from topgrade
I can successfully start the Windows update process via the GUI with no issues.

## Configuration file (Optional)
this is a vanilla installation with a vanilla configuration file.

```toml

```

## Additional Details
- Operation System/Version
Windows 11

- Installation
installed through scoop

- Topgrade version (`topgrade -V`)

## Verbose Output (`topgrade -v`)
this is the portion where the problem araises.

── 05:40:34 - Windows Update ───────────────────────────────────────────────────
Running Windows Update. Check the control panel for progress.
DEBUG Executing command `C:\WINDOWS\system32\UsoClient.EXE ScanInstallWait`
DEBUG Command failed: Err(
   0: Command failed: `C:\WINDOWS\system32\UsoClient.EXE ScanInstallWait`
   1: `C:\WINDOWS\system32\UsoClient.EXE` failed: exit code: 0x80070057

Location:
   src\steps\os\windows.rs:157

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.)
DEBUG Step "Windows update" failed:
   0: Command failed: `C:\WINDOWS\system32\UsoClient.EXE ScanInstallWait`
   1: `C:\WINDOWS\system32\UsoClient.EXE` failed: exit code: 0x80070057

Location:
   src\steps\os\windows.rs:157

**Comment by bb010g** [comment]
Related to topgrade-rs/topgrade#345, an issue about UsoClient specifically failing on Windows 11. This issue is about topgrade more generally failing to invoke Windows Update on Windows 11.

**Comment by SteveLauC** [comment]
My VM is on Windows 10 Pro 22H2, I got no outputs either by involving the command directly or using `Command::new("UsoClient").arg("ScanInstallWait")` in Rust code...

**Comment by niStee** [comment]
#94 maybe this feature as a workaround in the meantime?

**Comment by SteveLauC** [comment]
> #94 maybe this feature as a workaround in the meantime?

Yeah, it can be seen as a workaround for this issue, but we should still try to find a usable way to do the system update on Windows

**Comment by niStee** [comment]
Implemented #671 as a working way to update the system on Windows 10/11. @GotAnAccount let me know if it works for you.

**Comment by endquote** [comment]
Doesn't work for me in an elevated powershell 7.4.1 prompt on a new machine with topgrade `14.0.1`. I have run `Set-ExecutionPolicy RemoteSigned` and `Install-Module -Name PSWindowsUpdate` but still get the error:

```
Import-Module : File C:\Users\jsantangelo\Documents\PowerShell\Modules\PSWindowsUpdate\2.2.1.4\PSWindowsUpdate.psm1
cannot be loaded because running scripts is disabled on this system.
```

**Comment by SteveLauC** [comment]
> I would just disable Windows Update but it doesn't look like there's a config option for that.

You can do this with disabling the system step

**Comment by endquote** [comment]
> > I would just disable Windows Update but it doesn't look like there's a config option for that.
> 
> You can do this with disabling the system step

Thanks — I was looking for it under the windows section.

**Comment by niStee** [comment]
#842

**Comment by SteveLauC** [comment]
Gentle ping on @GotAnAccount, do you want to try the Topgrade with the latest commit and see if this issue is resolved now?

**Comment by GotAnAccount** [comment]
I've been out of the loop. So I did the update and the error disapeared, but now I simply get this instead:

── 07:16:55 - Windows Update ───────────────────────────────────────────────────
Consider installing PSWindowsUpdate as the use of Windows Update via USOClient is not supported.

And I installed the module using this command:

Install-Module -Name PSWindowsUpdate -AllowClobber -Force

**Comment by SteveLauC** [comment]
Does it work after installing the module?

**Comment by GotAnAccount** [comment]
No, it did not.

**Comment by SteveLauC** [comment]
> No, it did not.

Could you please provide more information, did it give an error?

**Comment by niStee** [comment]
@GideonBear I suggest we close this for now, and if the bug reappears, i.e. we get detailed feedback, we can reopen this issue.

**Comment by GideonBear** [comment]
Agreed!
The original issue is about `UsoClient`, which is not used by Topgrade anymore. @GotAnAccount if your issue persists, please open a new issue.
